### DISCUSSION TITLE
Victron VE.Direct Kabel - TX/RX kreuzen?

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#534] [Q&A] [answered]
Ich habe eine Frage: Was macht am am häufigsten?
Eine Victron Doku zeigt das Ve.Direkt Kabel "Direkt", eine andere "gekreuzt" = Die PINS 1-4 sind richtig nur auf der Linken Seite.
Ich persoenlich benutze ich Direkt weil ich meine Kabel selber baue und meine Freunde wenig Ahnung von gekreuzten Kabeln verstehen.
Ich wurde aber gerne wissen, was andere machen... 
![Connection Straight vs Crossed](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/4b581b3b-d43e-4e85-8c38-8d4eb2090303)

![17835-vedirect-documentation](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/22bc8a63-cf6a-4d50-859d-b2df50908e69)

![VICTRON PINS2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/e7b9e257-693b-4b89-ab52-6d6743476725)

**Comment by spcqike** [comment]
ich verstehe deine Frage nicht und sehe keinen Unterschied in deinen Skizzen.
RX auf TX und umgekehrt, so wie du es in beiden Fällen hast. bei deinem "gekreuzten" Bild, sind beim Consumer TX und RX eben anders angeordnet. das Ergebnis (RX zu TX) ist bei beiden identisch.

**Reply by Manos1966** [reply]
Ja, vergleiche aber jetzt die PIN-Nummerierung des Producer und des Consumer.

Wichtiger aber, wie hast du dein Kabel gelegt?

**Comment by schlimmchen** [comment]
Es gibt hier keine Auswahl oder Option, es funktioniert nur so wie in der VE.Direct Dokumentation angeben: TX Pin des Producers (MPPT Charge Controller) zum RX Pin des Consumers (OpenDTU-OnBattery), sowie RX Pin des Producers zum TX Pin des Consumers.

Wo kommt folgendes her?

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/3578416/d3ef4801-8d92-4b8e-8251-0c40fa3e6280)

Das sieht mir nach Unfug aus.

Wenn du dich fragst, wie die VE.Direkt Kabel aufgebaut sind, so verlasse dich auf gar keinen Fall auf die Farben. An meinem Kabel war GND rot und VCC schwarz. Und am einen Ende des Kabels war die eine Farbe TX und am anderen die andere Farbe. Das macht auch Sinn, weil das Kabel für zwei VE.Direct Geräte gedacht war, sodass die Pins 2 und 3 gekreuzt belegt waren.

**Reply by Manos1966** [reply]
> Wo kommt folgendes her?

Victron Community

> The original Victron VE.Direct cables are crossed. 
With the firmware v2.60 partially (for example with the SmartShunt) straight VE.Direct cables worked. 
From firmware V2.61 only crossed VE.Direct cables work. The information in the document "VE.Direct Protocol" in the current version 3.30 in the section "Physical interface" is wrong.

[VE.Direct diy self made cables ](https://community.victronenergy.com/questions/48562/vedirect-diy-cable.html)

**Reply by swingstate** [reply]
Im Fall des onBattery PCBs habe ich es auch wie oben dargestellt implementiert und es funktioniert. @Manos1966 bezogen auf unsere Diskussion heute würde ich es dann auch dabei belassen und GND & VCC nicht kreuzen.

**Comment by 88markus88** [comment, marked as answer]
Immer Tx auf Rx und umgekehrt. Völlig unabhängig davon, wie die Pinbelegung ist und wie die Kabelfarben sind. 
Wichtig auch noch die galvanische Trennung (ADUM 1201 oder ähnliches) und die Pegelanpassung zwischen ESP und Victron Gerät. Macht der ADUM praktischerweise gleich mit. Direkt durchgeführte Kabel funktionieren sicher oft, können aber zu unreproduzierbaren Problemen führen. Den Ärger würde ich mir für den geringen Aufwand nicht antun.

**Comment by Manos1966** [comment]
OK, dann werde ich die Wiki dementsprechend anpassen wie folgt:
![VE Direct-documentation - Crossed](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/53d724ed-f586-41cb-92b4-9ea2f3e4cf1c)

**Reply by Manos1966** [reply]
Hier:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/VE.Direct-Interface#cables

**Comment by martingro** [comment]
Hier jetzt die Auflösung für doofe wie mich:
<img width="4032" height="3024" alt="image" src="https://github.com/user-attachments/assets/e0642e5c-3d98-463c-82fa-5665ee79ec43" />

PIN 1-4 am victron von links nach rechts (Pin 1 GND, 0 Ohm zu minus an den Pv/Battery klemmen) korrespondieren zu Pin 1-4 ab GND auf dem Can shield. Zusätzlich MUSS Pin 4 verbunden sein, da der Pegelwandler die 5V für den richtigen Pegel auf der Seite zum Victron braucht. Am Ende ist es offensichtlich, aber ich habe zu oft in anderen Tutorials gelesen, dass man die 5V nicht verbinden sollte.
